velox/functions/sparksql/RegexFunctions.cpp

```diff
@@ -110,60 +110,65 @@ template <typename T>
 struct RegexpReplaceFunction {
   VELOX_DEFINE_FUNCTION_TYPES(T);
 
-  void call(
+  bool callNullable(
       out_type<Varchar>& result,
-      const arg_type<Varchar>& stringInput,
-      const arg_type<Varchar>& pattern,
-      const arg_type<Varchar>& replace) {
-    re2::RE2* patternRegex = getCachedRegex(pattern.str());
-    re2::StringPiece replaceStringPiece = toStringPiece(replace);
-
-    std::string string(stringInput.data(), stringInput.size());
-    RE2::GlobalReplace(&string, *patternRegex, replaceStringPiece);
-
-    if (string.size()) {
-      result.resize(string.size());
-      std::memcpy(result.data(), string.data(), string.size());
-    } else {
-      result.resize(0);
-    }
+      const arg_type<Varchar>* stringInput,
```

Comment:
Would you check if documentation needs updating to clarify the behavior for input NULLs and explain when result might be NULL?